### PR TITLE
feat(membership request): create entity

### PR DESCRIPTION
**Pull request by Defferrard**
This pull request is a part of the main issue https://github.com/graasp/graasp/issues/1220.
I've divided it, to make it easier to review and having continuous feedback

The goal of this PR is only to create the corresponding Entity and the structure of the feature. This PR should have no impact on the application.

### CODE REVIEW

File: src/services/itemMembership/plugins/MembershipRequest/entities/MembershipRequest.ts
Comment (ReidyT) [inline src/services/itemMembership/plugins/MembershipRequest/entities/MembershipRequest.ts:1]: Isn't the permission missing in this entity?

File: src/services/itemMembership/plugins/MembershipRequest/entities/MembershipRequest.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  PrimaryGeneratedColumn,
+  Unique,
+  UpdateDateColumn,
+} from 'typeorm';
+
+import { Item } from '../../../../item/entities/Item';
+import { Member } from '../../../../member/entities/member';
+
+@Entity()
+@Unique('UQ_membership_request_item-member', ['item', 'member'])
```
Comment (ReidyT) [inline src/services/itemMembership/plugins/MembershipRequest/entities/MembershipRequest.ts:16]: I really like the fact that we are naming the constraints and foreign key ! Do you think it is possible to also name PK and Idx ?

File: src/services/itemMembership/plugins/MembershipRequest/schemas.ts
Comment (ReidyT) [inline src/services/itemMembership/plugins/MembershipRequest/schemas.ts:1]: I think the permission is missing here too ? Or it is the admin who decide the permission ?

**Review by ReidyT — APPROVED** [review]
Great work ! LGTM! I just let some questions 😶‍🌫️

File: src/services/itemMembership/plugins/MembershipRequest/schemas.ts
Comment (ReidyT) [reply src/services/itemMembership/plugins/MembershipRequest/schemas.ts:1]: Perfect, I think it is a really good decision! Maybe we just have to update it in the #1220 issue ?

File: src/services/itemMembership/plugins/MembershipRequest/entities/MembershipRequest.ts (hunk 1 above)
Comment (Defferrard) [reply src/services/itemMembership/plugins/MembershipRequest/entities/MembershipRequest.ts:16]: Yes we can, I'll name the PK!

**Review by pyphilia — APPROVED** [review]
Nice PR, I just have a question about a column. Otherwise I think it's good to divide the work in many PRs, however since we deploy on dev on merge, you will have to write migrations if you make any change in between. Or we manually down the previous migration, which also a solution (we usually do it if we tested the migration before merging).

File: src/services/itemMembership/plugins/MembershipRequest/entities/MembershipRequest.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  })
+  @JoinColumn({ name: 'item_id', foreignKeyConstraintName: 'FK_membership_request_item_id' })
+  item: Item;
+
+  @CreateDateColumn({ name: 'created_at', nullable: false })
+  createdAt: Date;
+
+  @UpdateDateColumn({ name: 'updated_at', nullable: false })
+  updatedAt: Date;
+}
```
[the remark this answers is not in the document]
Comment (Defferrard) [reply src/services/itemMembership/plugins/MembershipRequest/entities/MembershipRequest.ts:40]: It was for the block feature. If we decide to set a state on the request. However, I agree it has no use right now so I can remove it !

**Pull request merged**